metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala

```diff
@@ -105,6 +111,22 @@ object MetalsEnrichments
       else
         None
     }
+
+    def getName(): String = {
+      if (buildTarget.getDisplayName == null) {
+        val name =
+          if (buildTarget.getBaseDirectory() != null)
+            buildTarget
+              .getId()
+              .getUri()
+              .stripPrefix(buildTarget.getBaseDirectory())
+          else
+            buildTarget.getId().getUri()
+
+        name.replaceAll("[^a-zA-Z0-9]+", "-")
```

Comment:
@tgodzik Should I change this so the non A->Z chars are _always_ replaced with `-`, not just when `DisplayName` is not supplied?
The issue is that in `FileDecoderProvider` Metals uses the display name to build up a URI to then display build target info to the user.  If there is a non URI valid character then this URI is built incorrectly.  You can see that issue also in https://github.com/scalameta/metals/pull/5816.
The other option would be to change the way build target info is displayed but that would probably involve changing each individual clients (vscode, vim etc) as well.